app/views/home/index.html.haml

```diff
@@ -57,20 +57,28 @@
           %img{:src => asset_path("check-outlined.svg")}/
           %p
             = benefit
-
-    %div.home-upload-ontology-button
-      = upload_ontology_button
+    .home-ontologies-actions
+      .home-upload-ontology-button
+        = upload_ontology_button
+      .discover-ontologies
+        = render Buttons::RegularButtonComponent.new(id: 'discover-ontologies-button', value: t('home.discover_ontologies_button'), variant: "secondary", state: "regular", href: "/ontologies") do |btn|
+          - btn.icon_right do
+            - inline_svg_tag "arrow-right.svg"
 
   .home-section
     %h4
       = t('home.recommender_annotator')
     %hr.home-section-line/
     .home-card
       %form{action: "/annotator_recommender_form", method: "post"}
-        %textarea.home-recommendations-and-annotations{rows: "6" , placeholder: t('home.paste_text_prompt'), name: "text"}
+        %textarea.home-recommendations-and-annotations{rows: "4" , placeholder: t('home.paste_text_prompt'), name: "text"}
         %input.d-none{type: "submit", name: "submit_button" ,value: "annotator", id: "annotator_submit"}/
         %input.d-none{type: "submit" , name: "submit_button" ,value: "recommender", id: "recommender_submit"}/
-        .home-services-buttons
+        .home-services-buttons.container
+          .insert-sample-text-button{onclick: "insertSampleText('#{t('sample_text')}')"}
```

Comment:
you can extract this into a stimulus controller, as you will re-use in multiple pages (annotator, recommended and homepage)